tests/integ/test_session.py

```diff
@@ -352,34 +351,46 @@ def test_table_exists(session):
-{Utils.random_name_for_temp_object(TempObjectType.TABLE)}"
-    assert session._table_exists(table_name) is False
-    session.sql(f'create temp table "{table_name}"(col_a varchar)').collect()
-    assert session._table_exists(table_name) is True
+    try:
+        table_name = f"{Utils.random_name_for_temp_object(TempObjectType.TABLE)}.\
+    {Utils.random_name_for_temp_object(TempObjectType.TABLE)}"
+        assert session._table_exists(table_name) is False
+        session.sql(f'create table "{table_name}"(col_a varchar)').collect()
+        assert session._table_exists(table_name) is True
+    finally:
+        session.sql(f'drop table if exists "{table_name}')
 
-    # handle single quotes
-    table_name = f"{Utils.random_name_for_temp_object(TempObjectType.TABLE)}..\
+    try:
+        table_name = f"{Utils.random_name_for_temp_object(TempObjectType.TABLE)}..\
     {Utils.random_name_for_temp_object(TempObjectType.TABLE)}. \
     {Utils.random_name_for_temp_object(TempObjectType.TABLE)}"
-    assert session._table_exists(f'"{table_name}"') is False
-    session.sql(f'create temp table "{table_name}"(col_a varchar)').collect()
-    assert session._table_exists(f'"{table_name}"') is True
+        assert session._table_exists(table_name) is False
+        session.sql(f'create table "{table_name}"(col_a varchar)').collect()
+        assert session._table_exists(table_name) is True
+    finally:
+        session.sql(f'drop table if exists "{table_name}')
+
+    # handle single quotes
+    try:
+        table_name = f"{Utils.random_name_for_temp_object(TempObjectType.TABLE)}..\
+        {Utils.random_name_for_temp_object(TempObjectType.TABLE)}. \
+        {Utils.random_name_for_temp_object(TempObjectType.TABLE)}"
+        assert session._table_exists(f'"{table_name}"') is False
+        session.sql(f'create table "{table_name}"(col_a varchar)').collect()
+        assert session._table_exists(f'"{table_name}"') is True
+    finally:
+        session.sql(f'drop table if exists "{table_name}')
 
     # handle double quotes to escape
-    table_name = f'""{Utils.random_name_for_temp_object(TempObjectType.TABLE)}..\
-    {Utils.random_name_for_temp_object(TempObjectType.TABLE)}. \
-    {Utils.random_name_for_temp_object(TempObjectType.TABLE)}""'
-    assert session._table_exists(f'"{table_name}"') is False
-    session.sql(f'create temp table "{table_name}"(col_a varchar)').collect()
-    assert session._table_exists(f'"{table_name}"') is True
+    try:
+        table_name = f'""{Utils.random_name_for_temp_object(TempObjectType.TABLE)}..\
+        {Utils.random_name_for_temp_object(TempObjectType.TABLE)}. \
+        {Utils.random_name_for_temp_object(TempObjectType.TABLE)}""'
+        assert session._table_exists(f'"{table_name}"') is False
+        session.sql(f'create table "{table_name}"(col_a varchar)').collect()
+        assert session._table_exists(f'"{table_name}"') is True
+    finally:
```

Comment:
I'm not sure if this will suppress assert failures?

Reply:
as discussed OOB assertion error will be caught with try-finally